File: code_blocks/integrations/third-party-integrations/firebase-integration_1.swift
import SwiftUI
import FirebaseAuth
import RevenueCat

@main
struct MyApp: App {
    init() {
        Purchases.configure(withAPIKey: "public_sdk_key")
        Auth.auth().addStateDidChangeListener { _, user in
            if let uid = user?.uid {
                Purchases.shared.logIn(uid) { _, _, error in
                    if let error {
                        print("Sign in error: \(error.localizedDescription)")
                    }
                }
            }
        }
    }

    var body: some Scene {
        WindowGroup {
            ContentView()
        }
    }
}

File: code_blocks/integrations/third-party-integrations/firebase-integration_1.dart
import 'package:firebase_auth/firebase_auth.dart';
import 'package:purchases_flutter/purchases_flutter.dart';

Future<void> initPlatformState() async {
  await Purchases.configure(PurchasesConfiguration('public_sdk_key'));
  FirebaseAuth.instance.authStateChanges().listen((user) {
    if (user != null) {
      Purchases.logIn(user.uid);
    }
  });
}
